tools/lsp/preview/native.rs

```diff
@@ -119,7 +119,22 @@ pub fn start_ui_event_loop(cli_args: crate::Cli) {
         }
     }
 
-    slint::run_event_loop_until_quit().unwrap();
+    let loop_result = slint::run_event_loop_until_quit();
+    if let Err(err) = loop_result {
+        let mut state_requested = GUI_EVENT_LOOP_STATE_REQUEST.lock().unwrap();
+        match *state_requested {
+            RequestedGuiEventLoopState::InitializationError(_)
+            | RequestedGuiEventLoopState::Uninitialized => unreachable!(),
+            RequestedGuiEventLoopState::QuitLoop => return,
+            RequestedGuiEventLoopState::StartLoop | RequestedGuiEventLoopState::LoopStarted => {
+                *state_requested = RequestedGuiEventLoopState::InitializationError(err.to_string());
+            }
+        }
+        GUI_EVENT_LOOP_NOTIFIER.notify_one();
+        while *state_requested != RequestedGuiEventLoopState::QuitLoop {
+            state_requested = GUI_EVENT_LOOP_NOTIFIER.wait(state_requested).unwrap();
```

Comment:
I'm curious, how does this work in this case?

If the loop was running (`RequestedGuiEventLoopState::LoopStarted`) `*state_requested` is set to `InitializationError`, then we call `notify_one()` on the condvar, then we wait in the line here... and who changes the state?

Reply:
This is running in the main (UI) thread.
The LSP thread is the one in control actually. And it is the LSP thread that should change the state to QuitLoop

Reply:
That's what I thought, too. But where is that happening? I couldn't find it.

Reply:
I mean, from this state, it will be blocked forever untill the LSP quits.
Further attempt to show the preview will all fail with the same error that was given before